app/views/pages/get-started/contact-us-tutorial/displaying-data.liquid

```diff
@@ -0,0 +1,228 @@
+
+  assign object = object | hash_merge: c
+
+  return object
+%}
+```
+{% endraw %}
+
+To adjust the check command for our _Contact Us_ form, let's simplify it as follows:
+
+#### app/lib/commands/contacts/create/check.liquid
+
+{% raw %}
+```liquid
+{% liquid
+  assign c = '{ "errors": {}, "valid": true }' | parse_json
+
+  assign object = object | hash_merge: c
+
+  return object
+%}
+```
+{% endraw %}
+
+In `app/lib/commands/contacts/create.liquid`, we have our `contact` object, which includes `email` and `body` properties. We then pass this object to the check command.
+
+In the check command, we created a new object `c`, which is short for `contact`.
+
+{% raw %}
+```liquid
+  assign c = '{ "errors": {}, "valid": true }' | parse_json
+```
+{% endraw %}
+
+This line initializes `c` with two properties:
+
+* `"errors"`:  a hash (another object) to store any validation errors.
+* `"valid"`: a boolean set to `true`, meaning the data is considered valid by default. If no validations are added, the validation will succeed.
+
+{% include 'alert/note', content: 'The `assign` tag creates a variable. Read the [assign tag documentation](https://documentation.platformos.com/api-reference/liquid/variables#assign) for further details.' %}
+
+Next, we merge this `c` object with our `contact` object using the `hash_merge` tag to combine the validation results with the original object:
+
+{% raw %}
+```liquid
+assign object = object | hash_merge: c
+```
+{% endraw %}
+
+{% include 'alert/note', content: '`hash_merge` takes two objects and merges them together. Read the [hash_merge filter documentation](/api-reference/liquid/platformos-filters#hash-merge) for further details.' %}
+
+This step combines the validation results with the original contact object. Now, the contact object includes both its original properties (`email` and `body`) and the validation results (`errors` and `valid`). This merged object allows us to track which fields failed validation and why in the logs.
+
+{% include 'alert/note', content: '`Parse_json` is a filter which transforms a string into a hash. Read the [Parse_json filter documentation](/api-reference/liquid/platformos-filters#parse-json)</code> for further details.' %}
+
+{% raw %}
+Filters are invoked via the "|" character. For example, `{{ 'hello' | upcase }}` invokes the `upcase` filter with "hello" as an argument. Tags, on the other hand, are invoked using `{% %}` characters.
+
+Keep in mind that the `parse_json` filter and tag are equivalent, and it's a matter of preference which one you use. If someone wants to associate the object `{ "hello": "world" }` with the variable "hello_json," they can achieve this with either the filter:
+{% endraw %}
+
+{% raw %}
+```liquid
+{% assign hello_json = '{ "hello": "world"}' | parse_json %}
+```
+{% endraw %}
+
+or using the tag:
+
+{% raw %}
+```liquid
+{% parse_json hello_json %}
+{ "hello": "world"}
+{% endparse_json %}
```

Comment:
@z-x are you thinking that some blocks can be just removed, or are you thinking about a general impression? Could you please write what you suggest to address the comment?

Reply:
I was thinking about the whole paragraph about `parse_json` tag vs filter. But since you've mentioned previously that the goal was to widely teach various aspects then maybe it's needed.

Might be personal, I just don't like sidequests in tutorials.